samples/posix/uname/README.rst

```diff
@@ -0,0 +1,56 @@
+.. _posix-uname-sample:
+
+POSIX uname()
+########################################################
+
+Overview
+********
+
+In this sample application, the POSIX uname() function
+is used to acquire system information and it is output to
+the console. Additionally, uname is added as a shell command and system
+information is displayed according to the option(s) provided for the command.
+
+
+Building and Running
+********************
+
+This project outputs to the console.  It can be built and executed
+on QEMU as follows:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/posix/uname
+   :host-os: unix
+   :board: qemu_x86
+   :goals: run
+   :compact:
+
+Sample Output
+=============
+
+.. code-block:: console
+
+    Printing everything in utsname...
+    sysname[7]: Zephyr
+    nodename[7]: zephyr
+    release[13]: 3.5.99
+    version[61]: zephyr-v3.5.0-3515-g10156f5f1d9c Jan  9 2024 22:23:04
+    machine[4]: x86
+
+
+    uart:~$ uname -a
+    Zephyr zephyr 3.5.99 zephyr-v3.5.0-3515-g10156f5f1d9c Jan  9 2024 22:23:04 x86 qemu_x86
+    uart:~$ uname -s
+    Zephyr
+    uart:~$ uname -n
+    zephyr
+    uart:~$ uname -r
+    3.5.99
+    uart:~$ uname -v
+    zephyr-v3.5.0-3515-g10156f5f1d9c Jan  9 2024 22:23:04
+    uart:~$ uname -m
+    x86
+    uart:~$ uname -i
+    qemu_x86
+    uart:~$ uname -smi
+    Zephyr x86 qemu_x86
```

Comment:
I don't think showing all the possible ways to run the sample is really necessary.
It certainly makes sense to have at least one with the multiple options though

```suggestion
    uart:~$ uname -a
    Zephyr zephyr 3.5.99 zephyr-v3.5.0-3515-g10156f5f1d9c Jan  9 2024 22:23:04 x86 qemu_x86
    uart:~$ uname -smi
    Zephyr x86 qemu_x86
```